Extension/src/LanguageServer/protocolFilter.ts

```diff
@@ -69,47 +61,49 @@ export function createProtocolFilter(): Middleware {
             }
         },
         didChange: async (textDocumentChangeEvent, sendMessage) => {
+            await clients.ActiveClient.awaitUntilLanguageClientReady();
             const me: Client = clients.getClientFor(textDocumentChangeEvent.document.uri);
             me.onDidChangeTextDocument(textDocumentChangeEvent);
-            sendMessage(textDocumentChangeEvent);
+            await sendMessage(textDocumentChangeEvent);
         },
-        willSave: defaultHandler,
+        willSave: invoke1,
         willSaveWaitUntil: async (event, sendMessage) => {
+            // await clients.ActiveClient.awaitUntilLanguageClientReady();
+            // Don't use awaitUntilLanguageClientReady.
+            // Otherwise, the message can be delayed too long.
             const me: Client = clients.getClientFor(event.document.uri);
             if (me.TrackedDocuments.has(event.document)) {
-                // Don't use me.requestWhenReady or notifyWhenLanguageClientReady;
-                // otherwise, the message can be delayed too long.
                 return sendMessage(event);
```

Comment:
Should this be `await sendMesssage(event)`. There's a known bug in which VS is not waiting for willSaveWaitUntil (https://github.com/microsoft/vscode/issues/164841).

Reply:
`await` is implied here, as the return value of the async lambda is a promise.  It would generate a lint warning/error if `await` is added, as it's considered superfluous.  (I had an `await` here and had to remove it.)

Based on the comment in the code, I left this message unchanged - it doesn't await the language client (and therefore anything else queued waiting for it), so this will still be delivered out of order (quickly).